integration_tests/e2e/changeSupportingPrison.cy.ts

```diff
@@ -0,0 +1,201 @@
+import PatientSearchPage from '../pages/changeSupportingPrison/patientSearch.page'
+import PatientSelectPage from '../pages/changeSupportingPrison/patientSelect.page'
+import SelectPrisonPage from '../pages/changeSupportingPrison/selectPrison.page'
+import ChangePrisonConfirmationPage from '../pages/changeSupportingPrison/changePrisonConfirmation.page'
+import ChangePrisonCompletedPage from '../pages/changeSupportingPrison/changePrisonCompleted.page'
+import Page from '../pages/page'
+
+const toOffender = $cell => ({
+  name: $cell[1].textContent,
+  prisonerNumber: $cell[2].textContent,
+  locationDescription: $cell[3].textContent,
+  supportingPrison: $cell[4].textContent,
+  changeSupportingPrisonLink: $cell[5].textContent,
+})
+
+context('Change supporting prison', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn', { roles: ['RESTRICTED_PATIENT_MIGRATION'] })
+    cy.task('stubManageUser')
+    cy.task('stubFrontendComponents')
+    cy.task('stubRestrictedPatientSearch', {
+      query: {
+        equalToJson: {
+          prisonerIdentifier: 'A1234AA',
+        },
+      },
+      results: {
+        content: [
+          {
+            cellLocation: '1-2-015',
+            firstName: 'JOHN',
+            lastName: 'SMITH',
+            prisonerNumber: 'A1234AA',
+            prisonName: 'HMP Moorland',
+            supportingPrisonId: 'MDI',
+            dischargedHospitalId: 'SHEFF',
+            dischargedHospitalDescription: 'Sheffield Hospital',
+            dischargeDate: '2021-08-11',
+          },
+        ],
+      },
+    })
+    cy.task('stubGetAgenciesByType', {
+      type: 'INST',
+      response: [
+        {
+          agencyId: 'MDI',
+          description: 'HMP Moorland',
+          longDescription: 'Moorland (HMP)',
+          agencyType: 'INST',
+          active: true,
+        },
+      ],
+    })
+    cy.task('stubGetAgencyDetails', {
+      id: 'MDI',
+      response: {
+        agencyId: 'MDI',
+        description: 'HMP Moorland',
+        longDescription: 'Moorland (HMP)',
+        agencyType: 'INST',
+        active: true,
+      },
+    })
+    cy.task('stubChangeSupportingPrison', {
+      status: 200,
+      response: {
+        restrictedPatient: {
+          supportingPrison: 'MDI',
+        },
+      },
+    })
+    cy.task('stubGetPrisonerDetails', {
+      prisonerNumber: 'A1234AA',
+      response: {
+        offenderNo: 'A1234AA',
+        firstName: 'JOHN',
+        lastName: 'SMITH',
+        locationDescription: 'Bartholemew Court',
+        categoryCode: 'C',
+        alerts: [
+          { active: true, alertType: 'T', alertCode: 'TCPA' },
+          { active: true, alertType: 'X', alertCode: 'XCU' },
+        ],
+      },
+    })
+    cy.signIn()
+  })
+
+  // it('should display the change supporting prison from the HomePage', () => {
```

Comment:
comment out the testing of the home page link for now as not integrated yet.  Will add in once testing is complete